JavaScript/JavaScriptSDK/AppInsights.ts

```diff
@@ -155,14 +155,40 @@ module Microsoft.ApplicationInsights {
          * Starts timing how long the user views a page or other item. Call this when the page opens. 
          * This method doesn't send any telemetry. Call {@link stopTrackTelemetry} to log the page when it closes.
          * @param   name  A string that idenfities this item, unique within this HTML document. Defaults to the document title.
+         * @param startDate A date that identifies the original start date that will override the current date. Defaults to the current date.
          */
-        public startTrackPage(name?: string) {
+        public startTrackPage(startDate?: Date); //Method firm with date
+        public startTrackPage(name?: string, startDate?: Date); //Method firm with name and date
+        public startTrackPage(param1?: (string | Date), param2?: Date) {
```

Comment:
This looks very hacky :) especially that the method with `param1` and `param2 is public and will be very confusing for SDK users. 

IMHO it would be better to only have one public method `public startTrackPage(name?: string)` which will internally call `startTrackPageInternal(name: string, startDate?: Date)`.  It will be an internal private method of `AppInsights` class. The snippet.js code can call `startTrackPageInternal` and pass a right set of parameters. The code may look like this: 

```
if (name.indexOf(trackPage) >= 0) {
   // change the function name from `startTrackPage` to `startTrackPageInternal` (the same for stopTrack)
   // check if `originalArguments` contains name, if not set it to `null`
   // [].push a call to startTrackPageInternal(name, new Date().getTime())
```

This way you will always have a name parameter passed to `startTrackPageInternal`. 

Thanks in advance for implementing this change!